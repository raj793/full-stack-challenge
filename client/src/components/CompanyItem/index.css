.company-item-container {
  display: flex;
  flex-direction: column;
  width: auto;
  height: 200px;
  color: #FFFFFF;
  border-bottom: 1px rgb(255, 255, 255, 0.4) solid;
}

.company-item-container:hover {
  opacity: 0.7;
  cursor: pointer;
}

.company-item-row {
  display: flex;
  flex-direction: row;
  height: 50%;
  align-items: center;
  padding-left: 20px;
  padding-right: 20px;
}

.company-item-divider {
  background-color: #FFFFFF;
  height: 50px;
  width: 2px;
  border-radius: 10px;
  margin-left: 20px;
  margin-right: 20px;
}

.company-item-name {
  font-family: Arial, Helvetica, sans-serif;
  font-size: 30px;
  align-items: center;
}