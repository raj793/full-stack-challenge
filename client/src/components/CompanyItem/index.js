import React from 'react';
import './index.css';

const CompanyItem = ({name, location, description, onClick}) => {

  const clickableFunc = () => {
    onClick && onClick();
  }

  return (
    <div className="company-item-container" onClick={clickableFunc}>
      <div className="company-item-row">
        <span className="company-item-name">{name}</span>
        <span className="company-item-divider" />
        <span>{location}</span>
      </div>
      <div className="company-item-row">
        {description}
      </div>
    </div>
  )
}

export default CompanyItem;