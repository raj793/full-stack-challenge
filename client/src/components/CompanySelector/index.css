.company-selector-container {
  display: flex;
  flex-direction: column;
  padding-left: 50px;
  padding-right: 50px;
}