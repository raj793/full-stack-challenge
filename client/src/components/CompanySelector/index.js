import React from 'react';
import { useHistory } from 'react-router-dom';

const CompanySelector = ({children}) => {
  const history = useHistory();
  const redirect = (id) => {
    history.push(`/company/${id}`);
  };

  return (
    React.Children.map(children, (child) => {
      if(React.isValidElement(child)) {
        return React.cloneElement(child, {
          onClick: () => redirect(child.props.id)
        })
      }
    })
  )
};

export default CompanySelector;