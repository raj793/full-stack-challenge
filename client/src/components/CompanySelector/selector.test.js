import React from 'react';
import renderer from 'react-test-renderer';
import CompanyItem from '../CompanyItem';
import CompanySelector from './index';

test('Selector Snapshot Test', () => {
  const formData = {
    name: "Name",
    city: "City",
    state: "State",
    description: "Description"
  }
  const component = renderer.create(
    <CompanySelector>
      <CompanyItem 
        name={formData.name} 
        location={`${formData.city} ${formData.state}`} 
        description={formData.description}
      />
    </CompanySelector>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});