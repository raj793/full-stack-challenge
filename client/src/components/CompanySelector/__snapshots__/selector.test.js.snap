// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector Snapshot Test 1`] = `
<div
  className="company-selector-container"
>
  <div
    className="company-item-container"
    onClick={[Function]}
  >
    <div
      className="company-item-row"
    >
      <span
        className="company-item-name"
      >
        Name
      </span>
      <span
        className="company-item-divider"
      />
      <span>
        City State
      </span>
    </div>
    <div
      className="company-item-row"
    >
      Description
    </div>
  </div>
</div>
`;
